### PR TITLE
CU-86dthj1c5 - Increasing SignMessage / VerifyMessage compatibility

### PR DESCRIPTION
**Pull request by melanke**

### CODE REVIEW

**Comment by melanke** [comment]
Task linked: [CU-86dthj1c5 Deep experiment with Dapi to check the difference between Neoline and NeonDappkit implementation](https://app.clickup.com/t/86dthj1c5)

File: packages/neon-dappkit/src/NeonSigner.ts [hunk 1]
```diff
@@ -74,7 +77,17 @@ export class NeonSigner implements Neo3Signer {
   }
 
   async verifyMessage(verifyArgs: SignedMessage): Promise<boolean> {
-    return wallet.verify(verifyArgs.messageHex, verifyArgs.data, verifyArgs.publicKey)
+    const messageHex = verifyArgs.message ?? u.hexstring2str(verifyArgs.messageHex)
+    return wallet.verify(messageHex, verifyArgs.data, verifyArgs.publicKey) || this.verifyMessageLegacy(verifyArgs)
```
Comment (melanke) [inline packages/neon-dappkit/src/NeonSigner.ts:81]: it looks like I introduced an undesired change here, let me check.

File: packages/neon-dappkit/NEON-SIGNER.md [hunk 2]
```diff
@@ -30,15 +30,17 @@ truly signed your specific message.
...
 
 // 3) check if the signature is valid, if the method returns true, it is certain that that specific publicKey signed that messageHex
 const valid = await signer.verifyMessage(mySignedMessage)
 ```
-You can use different **versions**, the default is `2`, but you can use `3` to sign a message without salt, and `1` to
-use the legacy version.
+You can use different **versions**:
+1. The same format as Neoline, with salt.
+2. The more readable version, easier to verify by other means (with salt).
+3. The same format as Neoline, without salt.
```
Comment (luc10921) [inline packages/neon-dappkit/NEON-SIGNER.md:44]: Isn't it better to use the `SignMessageVersion` enum instead?

Comment (melanke) [reply packages/neon-dappkit/NEON-SIGNER.md:53]: Done

**Pull request merged**